src/api/children.js

```diff
@@ -7,19 +7,12 @@ define(function () {
       return node;
     }
 
-    var treeWalker = document.createTreeWalker(node, NodeFilter.SHOW_ALL, null, false);
-    var previousNode = treeWalker.currentNode;
-    if (treeWalker.firstChild()) {
-      // TODO: build list of non-empty elements (used elsewhere)
-      // Do not include non-empty elements
-      if (treeWalker.currentNode.nodeName === 'BR') {
-        return previousNode;
-      } else {
-        return firstDeepestChild(treeWalker.currentNode);
-      }
-    } else {
-      return treeWalker.currentNode;
+    var child = node.firstChild;
+    if( child.nodeName === 'BR' ) {
+      return node;
     }
+
+    return firstDeepestChild(child);
```

Comment:
I think where TreeWalker really shines is where you have a complex DOM and only want to browser through specific nodes. In this case, where only `firstChild` is accessed, the cost of creating the TW is just too much. See a similar test case here: https://jsperf.com/treewalker-vs-dom-traversal or my proposal for a faster implementation of `isEmptyInlineElement` https://jsperf.com/node-vs-treewalker-apis